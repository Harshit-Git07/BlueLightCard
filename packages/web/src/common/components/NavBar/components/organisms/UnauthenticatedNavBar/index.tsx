import React from 'react';

const UnauthenticatedNavBar = () => {
  return <div data-testid="unauthenticated-navbar">UnauthenticatedNavBar</div>;
};

export default UnauthenticatedNavBar;
