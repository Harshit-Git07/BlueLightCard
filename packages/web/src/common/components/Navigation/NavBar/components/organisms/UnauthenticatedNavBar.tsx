import React, { useState } from 'react';
import { UnauthenticatedNavBarProps } from '../../types';
import Logo from '@/components/Logo';
import { BRAND } from '@/global-vars';
import MobileNavToggleButton from '../atoms/MobileNavToggleButton';
import Button from '@bluelightcard/shared-ui/components/Button-V2';
import { ThemeVariant } from '@bluelightcard/shared-ui/types';
import { BRANDS } from '@/root/src/common/types/brands.enum';
import DDSLogos from '../molecules/DDSLogos';
import MobileNavigation from './MobileNavigation';
import { NavBar } from '@bluelightcard/shared-ui';

const UnauthenticatedNavBar = ({ navigationItems, isSticky }: UnauthenticatedNavBarProps) => {
  const [showMobileMenu, setShowMobileMenu] = useState(false);

  const onShowMobileMenuClick = () => {
    setShowMobileMenu(!showMobileMenu);
  };

  return (
    <div data-testid="unauthenticated-navbar">
      <div
        className={`dark:bg-NavBar-bg-colour-dark bg-NavBar-bg-colour text-NavBar-item-text-colour dark:text-NavBar-item-text-colour-dark ${
          isSticky && !showMobileMenu
            ? 'border-b border-b-colour-onSurface-outline dark:border-b-colour-onSurface-outline-dark'
            : ''
        }`}
      >
        <div className="h-[72px] px-4 tablet:px-14 laptop:container laptop:mx-auto flex justify-between items-center">
          <Logo className="h-[35px] tablet:h-[42px] laptop:h-[55px]" url="/members-home" />
          <div className="flex gap-7 h-full items-center">
            <div className="hidden desktop:flex desktop:h-full">
              <NavBar links={navigationItems} />
            </div>
            <div className="flex gap-2 items-center">
              <div className="hidden tablet:flex tablet:gap-2 tablet:items-center">
                <div className="hidden tablet:flex desktop:hidden">
                  {(BRAND as BRANDS) === BRANDS.DDS_UK && <DDSLogos />}
                </div>
                <Button variant={ThemeVariant.Secondary} size="Large">
                  Log in
                </Button>
                <Button size="Large">Sign up</Button>
              </div>
              <div className="hidden desktop:flex">
                {(BRAND as BRANDS) === BRANDS.DDS_UK && <DDSLogos />}
              </div>
              <MobileNavToggleButton
                onIconClick={onShowMobileMenuClick}
                isMenuOpen={showMobileMenu}
              />
            </div>
          </div>
        </div>
        <div className="h-[72px] hidden tablet:px-14 tablet:flex desktop:hidden desktop:mx-auto">
          <NavBar links={navigationItems} />
        </div>
      </div>
      {showMobileMenu && (
        <MobileNavigation
          navigationItems={navigationItems}
          unauthenticatedProps={{ brand: BRAND }}
        />
      )}
    </div>
  );
};

export default UnauthenticatedNavBar;
