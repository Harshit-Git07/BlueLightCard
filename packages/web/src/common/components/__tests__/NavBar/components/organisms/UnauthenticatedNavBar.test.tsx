import { render } from '@testing-library/react';
import UnauthenticatedNavBar from '../../../../NavBar/components/organisms/UnauthenticatedNavBar';

describe('UnauthenticaedNavBar', () => {
  it('renders', () => {
    const { container } = render(<UnauthenticatedNavBar />);
    expect(container).toMatchSnapshot();
  });
});
