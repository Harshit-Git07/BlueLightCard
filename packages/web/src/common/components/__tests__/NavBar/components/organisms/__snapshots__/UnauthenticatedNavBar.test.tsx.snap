// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnauthenticaedNavBar renders 1`] = `
<div>
  <div
    data-testid="unauthenticated-navbar"
  >
    UnauthenticatedNavBar
  </div>
</div>
`;
